extensions/notification/NotificationConfigManager.js

```diff
@@ -1,283 +1,87 @@
-                    this._cachedConfigs.add(change.documentKey._id, bucketNotificationConfiguration);
-                    onConfigManagerCacheUpdate('update');
-                }
-                break;
-            default:
-                this._logger.debug('Skipping unsupported change stream event', {
-                    method: 'NotificationConfigManager._handleChangeStreamChange',
-                });
-                break;
+    getConfig(bucket, cb) {
+        const val = this._configManagerBackend.getConfig(bucket);
+        if (!cb) {
+            return val;
         }
-        this._logger.debug('Change stream event processed', {
-            method: 'NotificationConfigManager._handleChangeStreamChange',
-        });
+        if (val instanceof Promise) {
+            return val.then(res => cb(null, res)).catch(err => cb(err));
+        }
+        return cb(null, val);
     }
 
     /**
-     * Initializes a change stream on the metastore collection
-     * Only document delete and update/replace operations are
-     * taken into consideration to invalidate cache.
-     * Newly created buckets (insert operations) are not cached
-     * as queue populator instances read from different kafka
-     * partitions and so don't need the configs for all buckets
-     * @returns {undefined}
+     * Add/update bucket notification configuration.
+     * Only supported in the Zookeeper backend.
+     *
+     * @param {String} bucket - bucket
+     * @param {Object} config - bucket notification configuration
+     * @return {boolean} - true if set
      */
-    _setMetastoreChangeStream() {
-        /**
-         * To avoid processing irrelevant events
-         * we filter by the operation types and
-         * only project the fields needed
-         */
-        const changeStreamPipeline = [
-            {
-                $match: {
-                    $or: [
-                        { operationType: 'delete' },
-                        { operationType: 'replace' },
-                        { operationType: 'update' },
-                    ]
-                }
-            },
-            {
-                $project: {
-                    '_id': 1,
-                    'operationType': 1,
-                    'documentKey._id': 1,
-                    'fullDocument._id': 1,
-                    'fullDocument.value.notificationConfiguration': 1
-                },
-            },
-        ];
-        this._metastoreChangeStream = new ChangeStream({
-            logger: this._logger,
-            collection: this._metastore,
-            pipeline: changeStreamPipeline,
-            handler: this._handleChangeStreamChangeEvent.bind(this),
-            throwOnError: false,
-            useStartAfter: semver.gte(this._mongoVersion, '4.2.0'),
-        });
-        // start watching metastore
-        this._metastoreChangeStream.start();
+    setConfig(bucket, config) {
+        if (this._usesZookeeperBackend) {
```

Comment:
not needed : implement the method in mongo backend as well, and just return false